tests/manage/z_cluster/cluster_expansion/test_node_expansion.py

```diff
@@ -18,52 +14,8 @@ class TestAddNode(ManageTest):
     Automates adding worker nodes to the cluster while IOs
     """
 
-    def test_add_node(self):
-        """
-        Test for adding worker nodes to the cluster while IOs
-        """
-        new_nodes = 3
-        if config.ENV_DATA["platform"].lower() in constants.CLOUD_PLATFORMS:
-            dt = config.ENV_DATA["deployment_type"]
-            if dt == "ipi":
-                machines = machine_utils.get_machinesets()
-                logger.info(
-                    f"The worker nodes number before expansion {len(node.get_worker_nodes())}"
-                )
-                for machine in machines:
-                    add_new_node_and_label_it(machine)
-                logger.info(
-                    f"The worker nodes number after expansion {len(node.get_worker_nodes())}"
-                )
-
-            else:
-                logger.info(
-                    f"The worker nodes number before expansion {len(node.get_worker_nodes())}"
-                )
-                if config.ENV_DATA.get("rhel_workers"):
-                    node_type = constants.RHEL_OS
-                else:
-                    node_type = constants.RHCOS
-                assert add_new_node_and_label_upi(
-                    node_type, new_nodes
-                ), "Add node failed"
-                logger.info(
-                    f"The worker nodes number after expansion {len(node.get_worker_nodes())}"
-                )
-
-        elif config.ENV_DATA["platform"].lower() == constants.VSPHERE_PLATFORM:
-            pytest.skip(
-                "Skipping add node in Vmware platform due to "
-                "https://bugzilla.redhat.com/show_bug.cgi?id=1844521"
-            )
-            # Issue to remove skip code https://github.com/red-hat-storage/ocs-ci/issues/2403
-            # logger.info(f'The worker nodes number before expansion {len(node.get_worker_nodes())}')
-            # if config.ENV_DATA.get('rhel_user'):
-            #     pytest.skip("Skipping add RHEL node, code unavailable")
-            # node_type = constants.RHCOS
-            # assert add_new_node_and_label_upi(node_type, new_nodes), "Add node failed"
-            # logger.info(f'The worker nodes number after expansion {len(node.get_worker_nodes())}')
-        ceph_cluster_obj = CephCluster()
-        assert ceph_cluster_obj.wait_for_rebalance(
-            timeout=3600
-        ), "Data re-balance failed to complete"
+    assert add_nodes(ocs_nodes=True, node_count=3)
```

Comment:
you should not call any functionality here in class definition directly! This is braking the logic even for collecting tests as you can see here:
https://github.com/red-hat-storage/ocs-ci/pull/3790/checks?check_run_id=1813671500

```
  _ ERROR collecting tests/manage/z_cluster/cluster_expansion/test_node_expansion.py _
  tests/manage/z_cluster/cluster_expansion/test_node_expansion.py:12: in <module>
      class TestAddNode(ManageTest):
  tests/manage/z_cluster/cluster_expansion/test_node_expansion.py:17: in TestAddNode
      assert add_nodes(ocs_nodes=True, node_count=3)
  ocs_ci/ocs/node.py:1411: in add_nodes
      machines = machine_utils.get_machinesets()
  ocs_ci/ocs/machine.py:449: in get_machinesets
      for machine in machinesets_obj.get()["items"]:
  ocs_ci/ocs/ocp.py:232: in get
      return self.exec_oc_cmd(command)
  ocs_ci/ocs/ocp.py:130: in exec_oc_cmd
      config.ENV_DATA["cluster_path"], config.RUN.get("kubeconfig_location")
  E   KeyError: 'cluster_path'
  ```

As class is processed even when collecting logs the things you would like to achieve should be done by fixture I guess so when you actually running against real environment then you can run any oc command which add_nodes actually does behind the scenes.

Reply:
Done